Tooltips

rsandISU opened the pull request.

hpeinar commented on public/planner/css/style.css:
```diff
@@ -9,6 +9,21 @@ body.dz-drag-hover {
+    width: fit-content;
+    position: relative;
+    background: #555555;
+    color: #CCCCCC;
```
The fonts size and color should match what's used in the UI in general.

rsandISU replied:
We tried to match the object count popup for the UI, is there an element in particular you would like the tooltip to match?
![image](https://user-images.githubusercontent.com/71036768/234979933-0bd59e62-c690-4057-9b26-5b47ca374025.png)

hpeinar replied:
Fair idea, but Object counter really isn't a default planner UI so I'd rather have the tooltips look like the menu items on top of the page. So a bit smaller font and white color. 
![Uploading Screenshot 2023-04-28 at 02.19.14.png…]()

hpeinar commented on public/planner/css/style.css:
```diff
@@ -9,6 +9,21 @@ body.dz-drag-hover {
+    padding-left: 8px;
+    padding-right: 8px;
+    padding-top: 1px;
+    padding-bottom: 1px; 
```
Could just use shorthand version `padding: 1px 8px`

hpeinar commented:
<img width="381" alt="Screenshot 2023-04-26 at 22 47 50" src="https://user-images.githubusercontent.com/490959/234687619-b0ce9b35-750a-4063-909d-3e20959de49a.png">
There seems to be a bug with the Tooltip when moving the object, the tooltip does not dissapear visually.

Also for non-building things (like roads, crops etc) there's no tooltip.

rsandISU commented on public/planner/js/engine/building.js:
```diff
@@ -120,10 +128,35 @@ Building.prototype.moveHighlight = function moveHighlight(noFill) {
 Building.prototype.mouseover = function mouseover(e) {
+
+    if (this.placed) {
+        const x = +this.sprite.attr('x');
```
Would it make more sense for the function to be in the Building class because there is a tooltip per building? Or is the best option the Board class?

hpeinar replied:
As reported above with a general comment, the tooltip is currently missing from all placeable objects. 
I assume this was not intended and it should should tooltip for everything and then it makes sense to have it in Board class because it is a general tooltip function not directly tied to buildings.

hpeinar commented:
If your exam period is finished and you'd be interested in some more coding, I have bunch of things that need to be on the new version of the planner. The new version is fully typed. 

As the new version isn't open source, we can agree to a paid work as well, probably in form of per feature not hourly. And I'm sure you'd learn a thing or two while doing it.

If you (or your friends) would be interested, let me know @ henrikpeinar@gmail.com

rsandISU commented:
I added the toggle to the menu. I believe I added it to the correct sections (not exportdata). Please let me know if there is anything else that needs to be added for this feature. Our class project is due 5/4/2023 at 12:00am central time, and our grade improves if we are able to merge our changes. We really appreciate how responsive you have been, it has helped us understand the code a lot better.

hpeinar commented:
I'll check it right now, there is one last thing though :)

You never added it to changelog in the planner/index.html and your name(s) to the credits section. Feel free to do so and link to your github if you want (as done for some others in the credit section).

hpeinar reviewed (approved):
The options menu seems to be working as expected. I'll approve this but wait until you've added the changelog and credits before merging.

rsandISU commented:
The credits are updated, I completely forgot about that until you mentioned it.

hpeinar commented:
Wouldn't have been necessary to copy the changelog header itself, but small thing :)

It's merged and deployed to live: https://stardew.info/planner/

Best of luck with the course grade!
